docs/design/Cephfs-RDR-ConsistencyGroup.md

```diff
@@ -0,0 +1,411 @@
+
+* RS (ReplicationSource): A VolSync resource designed to
+  manage the source. Each RS is responsible for managing one PVC as the data source.
+* RGS (ReplicationGroupSource): A Ramen resource designed to
+  manage all RS instances in a consistency group.
+* RD (ReplicationDestination): A VolSync resource intended
+  for managing the destination. Each RD handles one PVC as the data destination.
+* RGD (ReplicationGroupDestination): A Ramen resource
+  designed to manage all RD instances in a consistency group.
+
+![RDR Enable](../diagrams/cephfs-cg-rdr-enable.drawio.svg)
+
+This diagram illustrates the design for enabling CephFS
+Regional Disaster Recovery (Regional-DR) with Consistency Group.
+To highlight the changes from this design to the current implementation,
+modifications have been appended to the bottom of the diagram
+rather than altering the original version.
+
+In step 6, currently, the PVCs in vrg are triaged into cephFS and RBD.
+To support consistency group, if a consistency group is specified,
+the cephFS PVCs also need to be further triaged into consistency group.
+
+In step 10, currently, Ramen creates a ReplicationSource for each PVC
+to enable Volsync to create independent volume snapshots for each PVC.
+Subsequently, Volsync restores each independent snapshot to a new
+read-only PVC to synchronize the data from it to the remote destination.
+
+In this design, since volsync doesn't currently support consistency groups,
+Ramen takes on the responsibility of creating volume group snapshots for
+the PVCs within the consistency group and restoring these snapshots to PVCs.
+Subsequently, Volsync synchronizes the data from the PVCs restored by
+Ramen to the remote destination.
+
+So, in step 10, there are some differences compared to the
+current implementation if a consistency group is specified:
+
+* Ramen get the existing VolumeGroupSnapshotClass
+* Ramen creates a ReplicationGroupSource with the schedule
+  specification from the primary VRG. Within each schedule:
+    * Ramen Update Status.LastSyncStartTime. Go to the Syncing state.
+      In Syncing state:
+        * Ramen creates a VolumeGroupSnapshot for the consistency group.
+          This VolumeGroupSnapshot generate VolumeSnapshots for each PVC
+          within the consistency group, all these VolumeSnapshots
+          within the same consistency group.
+        * Each snapshot is then restored to a new read-only PVC by Ramen
+        * Ramen creates a ReplicationSource (Use Manual and Direct
+          method in ReplicationSource) for each restored PVC
+        * Ramen monitors the manual string in the ReplicationSource
+          spec and status to ensure all PVCs in the group are
+          successfully synced to the remote destination.
+        * Once synchronization is completed, update Status.LastSyncStartTime
+          to nil, and Status.LastSyncTime. Go to Cleanup state
+    * In Cleanup state:
+        * Ramen cleans up each RS and the read-only PVCs
```

Comment:
Should not cleanup each RS. Each RS should stay around maintaining its status information until the next successful sync. In other words, once the RS for a PVC is created, it should stay around forever, just the sourcePVC and the trigger string should change with every new sync.